### ISSUE TITLE
Weight Decay configuration for BERT?

### ISSUE THREAD

**Issue by jeswan** [nyu-mll/jiant-v1-legacy#937]
<a href="https://github.com/zphang"><img src="https://avatars1.githubusercontent.com/u/1668462?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [zphang](https://github.com/zphang)**
_Thursday Oct 17, 2019 at 18:14 GMT_
_Originally opened as https://github.com/nyu-mll/jiant/issues/937_

----

The Transformers/pytorch_pretrained_bert version of the BERT training code applies weight decay to only a subset of the parameters: https://github.com/huggingface/transformers/blob/0856a231c0d6f3abb193e99ecf52ff76edb75fb8/examples/run_classifier.py#L482-L509

(I am referencing pytorch_pretrained_bert v0.6 because AllenNLP uses BertAdam from that version.)

As far as I can tell, we do not have the equivalent setup in jiant. I believe the current setup applies the default weight decay of 0.01 (which is the correct weight decay for the parameters that should have it applied) to all parameters.

Given that we have been able to train performant models with the current setup, I'm not sure if this qualifies as a bug or if we are satisfied with this arrangement.
